gcm_filters/filter.py

```diff
@@ -271,3 +318,22 @@ def apply(self, field, dims):
             dask="parallelized",
         )
         return field_smooth
+
+    def apply_to_vector(self, ufield, vfield, dims):
+        """Filter a vector field across the dimensions specified by dims."""
+
+        filter_func_vec = _create_filter_func_vec(self.filter_spec, self.Laplacian)
```

Comment:
We should be able to raise an informative error if a user tries to create to do a vector filter with a scalar laplacian I think we can resolve this by implementing the suggestion to have a separate `BaseScalarLaplacian` and `BaseVectorLaplacian` and then checking the type in the `Filter.__post_init__` method.

Reply:
Here is an attempt to raise an error based on the two base classes:
https://github.com/NoraLoose/gcm-filters/blob/59bb142a8c01224138d00026000a1732873271de/gcm_filters/filter.py#L330-L334

Does this look good to you?